Source/WebCore/platform/Theme.h

```diff
@@ -51,7 +53,7 @@ class Theme {
     // position cannot be determined by examining child content. Checkboxes and radio buttons are examples of
     // controls that need to do this. The adjustment is an offset that adds to the baseline, e.g., marginTop() + height() + |offset|.
     // The offset is not zoomed.
-    virtual int baselinePositionAdjustment(StyleAppearance) const;
+    virtual int baselinePositionAdjustment(StyleAppearance, WritingMode) const;
```

Comment:
if you don't actually use block direction information, this may just be a isHorizontal/isVertical bit.